### PR TITLE
add tls e2e spec of deploying tls cluster

### PR DESCRIPTION
**Pull request by just1900**
### What problem does this PR solve?
Add tls e2e spec of deploying tls cluster

- [x] add cross kubernetes tls cluster deploy test
- [x] add tls server test
- [ ] add tls mysql client test

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
add e2e spec of deploying tls cluster cross kuberentes
```

### CODE REVIEW

**Review by handlerww — APPROVED** [review]

File: tests/e2e/tidbcluster/across-kubernetes.go [hunk 1]
```diff
@@ -157,72 +157,86 @@ var _ = ginkgo.Describe("[Across Kubernetes]", func() {
...
-			resp, err := pdutil.GetMembersV2(pdAddr)
-			framework.ExpectNoError(err, " failed to get pd members of cluster-1 %s/%s", tc1.Namespace, tc1.Name)
-			for _, member := range resp.Members {
-				peerURLs := []string{}
-				for _, url := range member.PeerURLs {
-					// url ex: http://cluster-1-pd-0.cluster-1-pd-peer.default.svc:2380
-					fields := strings.Split(url, ":")
-					fields[1] = fmt.Sprintf("%s.%s", fields[1], clusterDomain)
-					peerURLs = append(peerURLs, strings.Join(fields, ":"))
+			caSecret, err := c.CoreV1().Secrets(ns1).Get(context.TODO(), fmt.Sprintf("%s-ca-secret", tcName1), metav1.GetOptions{})
```
Comment (csuzhangxc) [inline tests/e2e/tidbcluster/across-kubernetes.go:178]: `var caSecret` before L171?

File: tests/e2e/tidbcluster/across-kubernetes.go [hunk 2]
```diff
@@ -157,72 +157,86 @@ var _ = ginkgo.Describe("[Across Kubernetes]", func() {
...
+
+			ginkgo.By("Ensure Dashboard use custom secret")
+			foundSecretName := false
+			// only check cluster-2 here, as cluster-1 is deployed the same as a normal tls-enabled tc.
+			pdSts, err := c.AppsV1().StatefulSets(ns2).Get(context.TODO(), controller.PDMemberName(tcName2), metav1.GetOptions{})
+			framework.ExpectNoError(err, "failed to get statefulsets for pd")
+			for _, vol := range pdSts.Spec.Template.Spec.Volumes {
+				if vol.Name == "tidb-client-tls" {
+					foundSecretName = true
+					framework.ExpectEqual(vol.Secret.SecretName, tc2DashTLSName)
```
Comment (csuzhangxc) [inline tests/e2e/tidbcluster/across-kubernetes.go:229]: `break` here?

**Review by csuzhangxc — APPROVED** [review]

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**